### PR TITLE
Update bfd.rst

### PR DESCRIPTION
**Pull request by maehdros**
Added explanation on the purpose of BFD

### CODE REVIEW

File: docs/configuration/protocols/bfd.rst [hunk 1]
```diff
@@ -9,6 +9,11 @@ BFD
 :abbr:`BFD (Bidirectional Forwarding Detection)` is described and extended by
 the following RFCs: :rfc:`5880`, :rfc:`5881` and :rfc:`5883`.
 
+In the age of very fast networks, a second of unreachability can means millions of packet loss. The idea behind BFD is to detect very quickly when a peer is down and take action extremely fast.
```
Comment (dmbaturin) [inline docs/configuration/protocols/bfd.rst]: This is some horrible wording. ;)

Something like "In the age of very fast networks, a second of unreachability may equal millions lost packets" would be much better.

File: docs/configuration/protocols/bfd.rst [hunk 2]
```diff
@@ -9,6 +9,11 @@ BFD
 :abbr:`BFD (Bidirectional Forwarding Detection)` is described and extended by
 the following RFCs: :rfc:`5880`, :rfc:`5881` and :rfc:`5883`.
 
+In the age of very fast networks, a second of unreachability can means millions of packet loss. The idea behind BFD is to detect very quickly when a peer is down and take action extremely fast. 
+
+For example without BFD, when a BGP or OSPF peer is unreachable but if the interface is still up it needs to wait for the timeout to expires to conludes that a peer is unreachable.
```
Comment (dmbaturin) [inline docs/configuration/protocols/bfd.rst]: "for the timeout to expire"

File: docs/configuration/protocols/bfd.rst [hunk 3]
```diff
@@ -9,6 +9,11 @@ BFD
...
+
+BFD uses small UDP packet very quickly to ensures that the peer is still alive.
```
Comment (dmbaturin) [inline docs/configuration/protocols/bfd.rst:18]: Please run the patch through grammarly.com or similar. The content is good, but the number of grammar mistakes seemingly coming from mere inattention is overwhelming.

**Review by dmbaturin — CHANGES_REQUESTED** [review]
When grammar mistakes are fixed, this should be a good explanation.

File: docs/configuration/protocols/bfd.rst (hunk 3 above)
Comment (maehdros) [reply docs/configuration/protocols/bfd.rst:18]: Ha ha Sorry, English is not my mother tongue. I am doing my best though ;)

File: docs/configuration/protocols/bfd.rst [hunk 4]
```diff
@@ -9,13 +9,13 @@ BFD
 :abbr:`BFD (Bidirectional Forwarding Detection)` is described and extended by
 the following RFCs: :rfc:`5880`, :rfc:`5881` and :rfc:`5883`.
 
-In the age of very fast networks, a second of unreachability can means millions of packet loss. 
+In the age of very fast networks, a second of unreachability may equal millions of lost packets.
 The idea behind BFD is to detect very quickly when a peer is down and take action extremely fast. 
 
-This allows to avoid the timers defined in BGP and OSPF protocol for example.
-Why wait for 30 (HOLDTIME) seconds if the peer is unable to communicate ?
+This allows avoiding the timers defined in BGP and OSPF protocol to expires.
```
Comment (dmbaturin) [inline docs/configuration/protocols/bfd.rst:15]: to expire

File: docs/configuration/protocols/bfd.rst [hunk 5]
```diff
@@ -9,13 +9,13 @@ BFD
...
+Why wait for 30 (HOLDTIME) seconds if the peer is unable to communicate?
 
-BFD uses small UDP packet very quickly to ensures that the peer is still alive.
+BFD uses smalls UDP packets very quickly to ensures that the peer is still alive.
```
Comment (dmbaturin) [inline docs/configuration/protocols/bfd.rst:18]: WTF. Just use a grammar checker, ok? grammarly.com will do.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: docs/configuration/protocols/bfd.rst (hunk 3 above)
Comment (dmbaturin) [reply docs/configuration/protocols/bfd.rst:18]: That's why I suggest that you should use grammarl.com

Comment (maehdros) [reply docs/configuration/protocols/bfd.rst:18]: Wel, I did follow your suggestion. But I guess that's not enough.

Comment (dmbaturin) [reply docs/configuration/protocols/bfd.rst:18]: I don't know what you did. If you simply accept its suggestions, most of the problems will be gone.

**Pull request merged**